src/banks/registries/directory.py

```diff
@@ -43,19 +50,42 @@ def _scan(self):
         self._index_path.write_text(self._index.model_dump_json())
 
     def get(self, *, name: str, version: str | None = None) -> "Prompt":
-        version = version or "0"
+        version = version or default_version
         for pf in self._index.files:
             if pf.name == name and pf.version == version and pf.path.exists():
                 return Prompt(pf.path.read_text())
         raise TemplateNotFoundError
 
     def set(self, *, name: str, prompt: Prompt, version: str | None = None, overwrite: bool = False):
-        version = version or "0"
+        version = version or default_version
         for pf in self._index.files:
             if pf.name == name and pf.version == version and overwrite:
                 pf.path.write_text(prompt.raw)
                 return
-        new_prompt_file = self._path / "{name}.{version}.jinja"
+        new_prompt_file = self._path / f"{name}.{version}.jinja"
         new_prompt_file.write_text(prompt.raw)
         pf = PromptFile(name=name, version=version, path=new_prompt_file)
         self._index.files.append(pf)
+
+    def get_meta(self, *, name: str, version: str | None = None) -> dict:
+        version = version or default_version
+        meta_path = self._meta_path / f"{name}.{version}.json"
+        if not meta_path.exists():
+            msg = f"Meta directory or file for prompt {name}:{version}.jinja not found."
+            raise FileNotFoundError(msg)
+        return json.loads(open(meta_path).read())
+
+    def set_meta(self, *, meta: dict, name: str, version: str | None = None, overwrite: bool = False):
+        version = version or default_version
+        if not self._meta_path.exists():
+            self._meta_path.mkdir()
+        if Path(self._path / f"{name}.{version}.jinja") not in [pf.path for pf in self._index.files]:
+            msg = f"Prompt {name}.{version}.jinja not found in the index. Cannot set meta for a non-existing prompt."
+            raise ValueError(msg)
+
+        if f"{name}:{version}.json" in self._meta_path.glob("*.json"):
```

Comment:
`glob` returns a generator so this should be something like
```
if f"{name}:{version}.json" in list(self._meta_path.glob("*.json"))
```
is it working?

Reply:
You are right, it wasn't. I fixed and added unit tests as well. :)
Thanks for catching this one :raised_hands: